platforms/cpuinfo.go

```diff
@@ -0,0 +1,79 @@
+package platforms
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"runtime"
+	"strings"
+)
+
+// For Linux, the kernel has already detected the ABI, ISA and Features.
+// So we don't need to access the ARM registers to detect platform information
+// by ourselves. We can just parse these information from /proc/cpuinfo
+func getCpuInfo(pattern string) (info string, err error) {
+	if !isLinuxOS(runtime.GOOS) {
+		return "", fmt.Errorf("Unsupported Operation System: %s", runtime.GOOS)
```

Comment:
Don't capitalize error message. Also, use errdefs and wrap here to return an error.

Reply:
Ok, I will use it in next version.